drivers/clock_control/Kconfig.nrf

```diff
@@ -47,14 +47,51 @@ config CLOCK_CONTROL_NRF_K32SRC_EXT_FULL_SWING
 
 endchoice
 
-config CLOCK_CONTROL_NRF_K32SRC_BLOCKING
-	bool "Blocking 32KHz crystal oscillator startup"
-	depends on CLOCK_CONTROL_NRF_K32SRC_XTAL
+config CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION
+	bool
+	depends on !SOC_SERIES_NRF91X
+	default y if CLOCK_CONTROL_NRF_K32SRC_RC
+
+if CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION
+
+config CLOCK_CONTROL_NRF_CALIBRATION_PERIOD
+	int "Calibration opportunity period (in 250ms units)"
+	default 1
+	range 1 127
+	help
+	  Periodically, calibration action is performed. Action includes
+	  temperature measurement followed by clock calibration. Calibration may
+	  be skipped if temperature change (compared to measurement of previous
+	  calibration) did not exceeded CLOCK_CONTROL_NRF_CALIBRATION_TEMP_DIFF
+	  and number of consecutive skips did not exceeded
+	  CLOCK_CONTROL_NRF_CALIBRATION_MAX_SKIP.
+
+config CLOCK_CONTROL_NRF_CALIBRATION_MAX_SKIP
```

Comment:
Do you really need this feature if the default CTIV is 16 (4 s) instead of 1 (0.25 s)?  I guess it does allow a maximum interval of 255*31.25 s, and allows rapid recalibration of the temperature changes at rates like 1 Cel / s but I'm not convinced the complexity of a two-level timeout counter is warranted.